### PR TITLE
feat(azuredevops): Create client, fetch project ids

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Related to https://github.com/cloudquery/cloudquery/issues/5511. This bootstrap the Azure client and manually adds a table with only one Id field.
The next PR will auto gen tables

### CODE REVIEW

File: plugins/source/azuredevops/client/client.go [hunk 1]
```diff
@@ -2,24 +2,45 @@ package client
...
 
 func New(ctx context.Context, logger zerolog.Logger, s specs.Source) (schema.ClientMeta, error) {
-	return &AzureDevOpsClient{}, nil
+	var spec Spec
+	err := s.UnmarshalSpec(&spec)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal GitHub spec: %w", err)
+	}
+
+	if spec.PersonalAccessToken == "" {
```
Comment (candiduslynx) [inline plugins/source/azuredevops/client/client.go:34]: I really think we should allow for env variables for keys.
@jsonpr could you share any insights on the security perspective for the keys hard-coded in the config?

File: plugins/source/azuredevops/client/client.go [hunk 2]
```diff
@@ -2,24 +2,45 @@ package client
...
 	"github.com/cloudquery/plugin-sdk/specs"
+	"github.com/microsoft/azure-devops-go-api/azuredevops"
 	"github.com/rs/zerolog"
 	"github.com/rs/zerolog/log"
 )
 
-type AzureDevOpsClient struct {
+type Client struct {
+	logger     zerolog.Logger
+	Connection *azuredevops.Connection
```
Comment (candiduslynx) [inline plugins/source/azuredevops/client/client.go:16]: just embed

File: plugins/source/azuredevops/resources/services/core/projects.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+func fetchProjects(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	cl := meta.(*client.Client)
+	coreClient, err := core.NewClient(ctx, cl.Connection)
+	if err != nil {
+		return err
+	}
+
+	stateFilter := &core.ProjectStateValues.All
+	projects, err := coreClient.GetProjects(ctx, core.GetProjectsArgs{StateFilter: stateFilter})
```
Comment (candiduslynx) [inline plugins/source/azuredevops/resources/services/core/projects.go:33]:
```go
input := core.GetProjectsArgs{StateFilter: &core.ProjectStateValues.All}
for {
	projects, err := coreClient.GetProjects
	if err != nil {
		return err
	}

	res <- projects.Value

	if len(projects.ContinuationToken) == 0 {
		break
	}

	input.ContinuationToken = &projects.ContinuationToken
}
```

**Review by candiduslynx — CHANGES_REQUESTED** [review]

File: plugins/source/azuredevops/client/client.go (hunk 2 above)
Comment (erezrokah) [reply plugins/source/azuredevops/client/client.go]: Done in https://github.com/cloudquery/cloudquery/pull/5611/commits/27323ab43af3261f7d522153912ef810eaf8fce5

File: plugins/source/azuredevops/resources/services/core/projects.go (hunk 3 above)
Comment (erezrokah) [reply plugins/source/azuredevops/resources/services/core/projects.go]: Done in https://github.com/cloudquery/cloudquery/pull/5611/commits/5434df748cdaf9964d9597e10985da2fd509c8a8

File: plugins/source/azuredevops/resources/services/core/projects.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
+package core
+
+import (
+	"context"
+
+	"github.com/cloudquery/cloudquery/plugins/source/azuredevops/client"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/core"
```
Comment (candiduslynx) [inline plugins/source/azuredevops/resources/services/core/projects.go:8]: https://pkg.go.dev/github.com/microsoft/azure-devops-go-api/azuredevops/v6

File: plugins/source/azuredevops/go.mod [hunk 5]
```diff
@@ -4,6 +4,7 @@ go 1.19
 
 require (
 	github.com/cloudquery/plugin-sdk v1.12.0
+	github.com/microsoft/azure-devops-go-api/azuredevops v1.0.0-b5
```
Comment (candiduslynx) [inline plugins/source/azuredevops/go.mod:7]: https://pkg.go.dev/github.com/microsoft/azure-devops-go-api/azuredevops/v6

**Review by candiduslynx — CHANGES_REQUESTED** [review]

File: plugins/source/azuredevops/go.mod (hunk 5 above)
Comment (erezrokah) [reply plugins/source/azuredevops/go.mod]: Good catch done in https://github.com/cloudquery/cloudquery/pull/5611/commits/0e33317e24f44fc4e466ef167b661b9147cd47a1.

I guess they don't bother updating the readme https://github.com/microsoft/azure-devops-go-api/tree/00dac5c867394a3c5ca4e12b6965d7625a1588c6#azure-devops-go-api

**Review by candiduslynx — APPROVED** [review]

**Pull request merged**